### PR TITLE
SRCH-4984 Allow regular search to return faceted results

### PR DESCRIPTION
**Pull request by stevenbarragan**
## Summary
- Move faceted search functionality from i14y_search_api to  i14y_search so regular search can return faceted results as well.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [ ] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [ ] You have specified at least one "Reviewer".

### CODE REVIEW

**Comment by luisgmetzger** [comment]
@stevenbarragan - Is there any tests that need to added or modified for this?

**Comment by stevenbarragan** [comment]
@luisgmetzger I added a test to make sure facet fields are making it to the react app.

File: app/models/api_search.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module ApiSearch
+  def initialize(options = {})
+    super
+
+    @highlight_options[:pre_tags]  ||= Api::V2::HighlightOptions::DEFAULT[:pre_tags]
+    @highlight_options[:post_tags] ||= Api::V2::HighlightOptions::DEFAULT[:post_tags]
```
Comment (luisgmetzger) [inline app/models/api_search.rb:8]: Not sure if we will be using it but could it be beneficial to set `attrs` here for the instantiated variables.

File: config/application.rb [hunk 2]
```diff
@@ -68,10 +68,6 @@ class Application < Rails::Application
     # Rails 5.0 had false.
     config.active_record.belongs_to_required_by_default = false
 
-    # Temporary workaround for:
-    # https://discuss.rubyonrails.org/t/cve-2022-32224-possible-rce-escalation-bug-with-serialized-columns-in-active-record/81017
```
Comment (luisgmetzger) [inline config/application.rb:72]: This doesn't look to be part of the ticket. Are we just good to remove from our earlier discussion this morning?

File: app/models/api_search.rb (hunk 1 above)
Comment (stevenbarragan) [reply app/models/api_search.rb:8]: I not sure why we would need to set attrs. It everything works without them I rather keep it like it is now.

File: config/application.rb (hunk 2 above)
Comment (stevenbarragan) [reply config/application.rb:72]: Yes. Removing this config since it's not doing anything anymore.

Comment (stevenbarragan) [reply config/application.rb:72]: Actually it still exists in rails. I'm going to put it back for now.

**Review by luisgmetzger — APPROVED** [review]

**Pull request merged**